s3fs/core.py

```diff
@@ -137,6 +138,12 @@ async def _error_wrapper(func, *, args=(), kwargs=None, retries):
         except Exception as e:
             err = e
     err = translate_boto_error(err)
+
+    s3 = func.__self__
+    if isinstance(err, PermissionError) and s3._client_config.signature_version == botocore.UNSIGNED:
```

Comment:
Were you able to test this? 

I am not sure if anon is guaranteed to be the same as `botocore.UNSIGNED` but as far as I could see it seems like it. It would be nice though if somebody with more experience with botocore can confirm this.

Reply:
I am suggesting 
```suggestion
    if isinstance(err, PermissionError) and self.anon:
```